app/src/main/java/app/revanced/integrations/settingsmenu/SettingsUtils.java

```diff
@@ -0,0 +1,52 @@
+package app.revanced.integrations.settingsmenu;
+
+import android.preference.Preference;
+import android.preference.PreferenceGroup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Class is used across multiple target apps.
+ *
+ * This entire class can _not_ reference:
+ * {@link app.revanced.integrations.settings.SettingsEnum}
+ * {@link app.revanced.twitch.settings.SettingsEnum}
+ * {@link app.revanced.tiktok.settings.SettingsEnum}
+ *
+ * or any other code that references these app specific integration code.
+ */
+public class SettingsUtils {
+
+    /**
+     * Sort the preferences by title and ignore the casing.
+     *
+     * Android Preferences are automatically sorted by title,
+     * but if using a localized string key it sorts on the key and not the actual title text that's used at runtime.
+     *
+     * @param menuDepthToSort Maximum menu depth to sort. Menus deep than this value will show it's preferences in the order created in patches.
+     */
+    public static void sortPreferenceGroupByTitle(PreferenceGroup group, int menuDepthToSort) {
+        if (menuDepthToSort == 0) return;
+
+        final int prefCount = group.getPreferenceCount();
+        List<Preference> preferences = new ArrayList<>(prefCount);
+        for (int i = 0; i < prefCount; i++) {
+            Preference preference = group.getPreference(i);
+            if (preference instanceof PreferenceGroup) {
+                sortPreferenceGroupByTitle((PreferenceGroup) preference, menuDepthToSort - 1);
+            }
+            preferences.add(preference);
+        }
+        Collections.sort(preferences, (pref1, pref2) ->
+                pref1.getTitle().toString().toLowerCase().compareTo(pref2.getTitle().toString().toLowerCase())
+        );
```

Comment:
This can be improved by mapping `preferences` to `preference.getTitle().toString().toLowerCase()`

Reply:
Instead of calling `pref1.getTitle().toString().toLowerCase()` inside the lambda, instead do:

`Collections.sort(preferences.map { it.getTitle().toString().toLowerCase() }, ...)`

Reply:
Then you can use a map of the preference to its ` getTitle().toString().toLowerCase().` Then you can compare that instead of having to call `getTitle().toString().toLowerCase()` inside the lambda